### PR TITLE
ojopaul:feature/#616_Author_Profile_URL

### PR DESCRIPTION
**Pull request by ojopaul**
## Description

Add View button link to Author Profile URL #616

## Benefits

A clickable link is now added to the front of slug field to enable easy click to view author page

## Applicable issues

https://github.com/publishpress/PublishPress-Authors/issues/616

## Checklist

- [x] I have created a specific branch for this pull request before committing, starting off the current HEAD of `development` branch. 
- [x] I'm submitting to the `development`, feature/hotfix/release branch. (Do not submit to the master branch!)
- [x] This pull request relates to a specific problem (bug or improvement).
- [x] I have mentioned the issue number in the pull request description text.
- [x] All the issues mentioned in this pull request relate to the problem I'm trying to solve.
- [x] The code I'm sending follows the [PSR-12](https://www.php-fig.org/psr/psr-12/) coding style.

### CODE REVIEW

File: src/assets/js/multiple-authors.js [hunk 1]
```diff
@@ -486,17 +486,27 @@ jQuery(document).ready(function ($) {
...
+        
+        /**
+         * Add view link to author url field
+         */
+         $('form#edittag tr.form-field #slug')
+             .css({
+                 'width': '85%',
+                 'margin-right': '5px'
+             })
+             .after('<a href="' + MultipleAuthorsStrings.home_url + '/author/' + $("#slug").val() + '/" class="button-secondary" target="_blank">' + MultipleAuthorsStrings.view_text + '</a>')
```
Comment (andergmartins) [inline src/assets/js/multiple-authors.js:509]: @ojopaul on a few sites, this line can be a problem. It creates the author URL in a way that will be invalid on sites that use plain permalinks.

The author URL should come from the object of `Author` class, `$author->link`. You can add it to the localize script array.

You can replicate it on a fresh WP site, or going to Settings > Permalinks > Plain. Save it and try to follow that View link.

**Review by andergmartins — CHANGES_REQUESTED** [review]

File: src/assets/js/multiple-authors.js (hunk 1 above)
Comment (ojopaul) [reply src/assets/js/multiple-authors.js:509]: Sorry, my thinking wasn't full when i was creating this feature including the css and end up doing rubbish. I was creating it in such a way that when user modify the permalink, new permalink will be the one opened and when i realised it won't be possible as the page will lead to 404 without saving first, i change end result without considering other closed options are now opened

Comment (ojopaul) [reply src/assets/js/multiple-authors.js]: @andergmartins, this PR is now updated

File: src/core/Plugin.php [hunk 2]
```diff
@@ -1382,6 +1384,21 @@ public function enqueue_scripts($hook_suffix)
...
+        if (
+            is_admin()
+            && $pagenow === 'term.php'
+            && isset($_GET['taxonomy']) && $_GET['taxonomy'] === 'author'
+            && isset($_GET['tag_ID'])
+        ) {
+            $author = Author::get_by_term_id((int)$_GET['tag_ID']);
+
+            if (is_object($author) && !is_wp_error($author) && isset($author->link)) {
+                $term_author_slug = $author->link;
```
Comment (andergmartins) [inline src/core/Plugin.php:1398]: The name `$term_author_slug` do not reflect the data the variable is storing. It stores the link, not the slug.

File: src/core/Plugin.php [hunk 3]
```diff
@@ -1429,7 +1446,7 @@ public function enqueue_scripts($hook_suffix)
                 'publishpress-authors'
             ),
             'mapped_author_nonce'           => wp_create_nonce("mapped_author_nonce"),
-            'home_url'                      => home_url(),
+            'term_author_slug'              => esc_url_raw($term_author_slug),
```
Comment (andergmartins) [inline src/core/Plugin.php:1449]: Same thing as above... the param do not stores the slug, but the link. It should have a more clear name.

**Review by andergmartins — CHANGES_REQUESTED** [review]

File: src/core/Plugin.php (hunk 2 above)
Comment (ojopaul) [reply src/core/Plugin.php]: Thanks @andergmartins, the PR is now updated.

**Comment by ojopaul** [comment]
@andergmartins, this PR is updated and ready for further review.

**Review by andergmartins — APPROVED** [review]

**Pull request merged**